fabric-entity-events-v1/src/main/java/net/fabricmc/fabric/api/entity/event/v1/ServerPlayerEvents.java

```diff
@@ -45,6 +46,21 @@
 		}
 	});
 
+	/**
+	 * An event that is called when a player takes fatal damage.
+	 *
+	 * <p>Mods can cancel this to keep the player alive.
+	 */
+	public static final Event<BeforeDeath> BEFORE_DEATH = EventFactory.createArrayBacked(BeforeDeath.class, callbacks -> (oldPlayer, damageSource, amount) -> {
+		boolean result = true;
+
+		for (BeforeDeath callback : callbacks) {
+			result &= callback.beforeDeath(oldPlayer, damageSource, amount);
```

Comment:
hmm, does the callback gets called if the result is already false? and should it be called in that case?

Reply:
They do, I was thinking about doing it either way but decided to do it this way (i.e. not short-circuiting) just to avoid surprising behaviour ('I registered this callback, why does it not always run?' etc.)

Reply:
It indeed does not short circuit, which is imo good enough. :+1: